realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -481,8 +503,12 @@ private boolean categorizeField(Element element) {
         return true;
     }
 
+    private boolean hasRequiredAnnotation(VariableElement field) {
```

Comment:
Perhaps add a comment here about why this method is different than `isRequiredField` and that one should normally be used.

Reply:
done